src/hooks/useMountedState.ts

```diff
@@ -31,16 +31,16 @@ import { useIsMounted } from './useIsMounted';
  * D is the type passed when then useMountedState is used
  * e.g. const [pass, setPass] = useMountedState<string>('someString');
  */
-export const useMountedState = <D>(initialValue: D) => {
+export const useMountedState = <D>(initialState: D | (() => D)) => {
   const isMounted = useIsMounted();
 
-  const [state, setState] = useState<D>(initialValue);
+  const [state, setState] = useState<D>(initialState);
 
   const setMountedState = (value: D) => {
     if (isMounted()) {
       setState(value);
     }
   };
```

Comment:
Please add `useCallback` here and I suspect this will solve your issue. Note that the leak is a separate thing - and also needs to be addressed.